File: Data Structures & Algorithms/binary-search/submission-2.java
public class Solution {
    public int search(int[] nums, int target) {
        int l = 0, r = nums.length;

        while (l < r) {
            int m = l + ((r - l) / 2);
            if (nums[m] > target) {
                r = m;
            } else {
                l = m + 1;
            }
        }
        return (l > 0 && nums[r - 1] == target) ? r - 1 : -1;
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-2.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:

        l=0
        
        if sum(nums) >=target :
            res = len(nums)
        else :
            return 0

        curr=0
        for r in range(l,len(nums)) :
            curr+=nums[r]
            if curr>=target :
                res=min(res,r-l+1)
                while curr>=target :
                    curr-=nums[l]
                    l+=1
                    if curr>=target :
                        res=min(res,r-l+1)
        return res


        